service/cloudconfig/render.go

```diff
@@ -0,0 +1,71 @@
+package cloudconfig
+
+import (
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_2_0_0"
+	"github.com/giantswarm/microerror"
+)
+
+func renderCloudProviderConfFile(params cloudProviderConfFileParams) (k8scloudconfig.FileAsset, error) {
+	fileMeta := k8scloudconfig.FileMetadata{
+		AssetContent: cloudProviderConfFileTemplate,
+		Path:         cloudProviderConfFileName,
+		Owner:        cloudProviderConfFileOwner,
+		Permissions:  cloudProviderConfFilePermission,
+	}
+
+	content, err := k8scloudconfig.RenderAssetContent(fileMeta.AssetContent, params)
+	if err != nil {
+		return k8scloudconfig.FileAsset{}, microerror.Mask(err)
+	}
+
+	file := k8scloudconfig.FileAsset{
+		Metadata: fileMeta,
+		Content:  content,
+	}
+
+	return file, nil
+}
+
+func renderGetKeyVaultSecretsFile(params getKeyVaultSecretsFileParams) (k8scloudconfig.FileAsset, error) {
+	fileMeta := k8scloudconfig.FileMetadata{
+		AssetContent: getKeyVaultSecretsFileTemplate,
+		Path:         getKeyVaultSecretsFileName,
+		Owner:        getKeyVaultSecretsFileOwner,
+		Permissions:  getKeyVaultSecretsFilePermission,
+	}
+
+	content, err := k8scloudconfig.RenderAssetContent(fileMeta.AssetContent, params)
```

Comment:
line 38,47 could be extracted eg:

```
func renderFiles(f k8scloudconfig.FileMetadata, p interface{}) (k8scloudconfig.FileAsset,erroe)
```
 and reused in all render function

Reply:
I would rather improve FileAsset and FileMeta instead.

Reply:
in a separate PR?

Reply:
This is part of k8scloudconfig. I don't promise I'll do that immediately. Also feel free to make a follow up PR if you find it important. I feel like adding `renderFiles` to every operator means the library should be improved.